packages/block-editor/src/components/url-input/index.js

```diff
@@ -305,7 +306,7 @@ class URLInput extends Component {
 					onInput={ stopEventPropagation }
 					placeholder={ placeholder }
 					onKeyDown={ this.onKeyDown }
-					onKeyPress={ this.props.onKeyPress || noop }
+					onKeyPress={ isFunction( __experimentalOnKeyPress ) ? __experimentalOnKeyPress : undefined }
```

Comment:
Not sure if there is a big performance gain but I thought it might be better not to define any callback rather than passing `noop`.

Reply:
So do we need the check at all then? Maybe just pass `__experimentalOnKeyPress` directly and then if it isn't provided then it will be `undefined` right?

Reply:
Sure that will do for case where you don't provide anything for the prop. However if you provide something else (like `false`), we'll allow an error if we don't check.

Reply:
gone